src/main/java/org/junit/Assert.java

```diff
@@ -812,6 +812,41 @@ public static void assertNotSame(Object unexpected, Object actual) {
         assertNotSame(null, unexpected, actual);
     }
 
+    /**
+     * Asserts that two {@link CharSequence} instances have the same contents.
+     * If they do not, an {@link AssertionError} is thrown.
+     *
+     * @param expected the expected contents
+     * @param actual the object to compare to
+     */
+    public static void assertContentEquals(String expected, CharSequence actual)  {
+        assertContentEquals(null, expected, actual);
+    }
+
+    /**
+     * Asserts that two {@link CharSequence} instances have the same contents.
+     * If they do not, an {@link AssertionError} is thrown with the given message.
+     *
+     * @param message the identifying message for the {@link AssertionError}
+     * @param expected the expected contents
+     * @param actual the object to compare to
+     */
+    public static void assertContentEquals(String message, String expected, CharSequence actual) {
+        if (expected == null && actual == null) {
+            return;
+        }
+        if (expected == null || actual == null) {
+            failNotEquals(message, expected, actual);
+        }
+
+        assertEquals(contentsDifferentMessage(message), expected.length(), actual.length());
```

Comment:
Moving the conversation about checking the length first to this pull request...

@sabi0 apparently suggested to check for the lengths to avoid the `toString()` call. I agree that failing early if the lengths differ with a message that just prints the lengths might not be all that helpful. We do, however, have one close precedent in JUnit, and that's when comparing arrays. If the lengths are different, we use a message like this:

`"<user supplied message>: array lengths differed, expected.length=2 actual.length=1; arrays first differed at element [1]; expected:<false> but was:<end of array>"`

(with no user-supplied message, the message starts with "array lengths...")

But I think `assertArrayEquals` does that because 1) arrays are often large, 2) the printed representation of arrays can be hard to read and 3) visually seeing the difference between two arrays in an error message is difficult. Here, we throw a `ComparisonFailure`, and the exception message generated by JUnit will show which parts of the string differ (plus IDEs often show a nice UI for `ComparisonFailure`). None of that applies here, so I think that even stating that the lengths are different in the exception isn't that useful for the method here.